rfcs/quick-start-book.md

```diff
@@ -0,0 +1,352 @@
+   3. Lack of polish should serve an end: don't show bad or sloppy practices without a good reason.
+   4. Showing how (and why!) to refactor your code is a very powerful teaching tool.
+6. Stick to a consistent style (e.g. for loops vs map) within each example.
+7. If you need to give advice that will only matter to some of your audience (e.g. how to handle an edge case, or support a specific platform), do so in a clearly marked aside or list.
+8. Examples should not use or rely on third-party plugins.
+These may be appropriate to link in "next steps" however at the end of the examples.
+   1. Third-party crates should be limited to the most essential, such as `rand`.
+
+## Implementation strategy
+
+1. The user-facing explanation above should live in `CONTRIBUTING.md` of the [`bevy-website` repo](https://github.com/bevyengine/bevy-website).
+2. This should be linked to from the website, likely under a "Contributing" tab.
+3. Per [bevy-website #143](https://github.com/bevyengine/bevy-website/issues/143),the website will need to be revamped somewhat.
+4. Automatic compilation of code snippets is the critical feature from the above, ensuring that we can keep the examples current in a reliable way.
+5. Once this RFC is merged, work should begin on a separate branch of `bevy-website` repo to create a new book.
+6. For the launch of 0.6 (or perhaps sooner), this branch is merged and becomes the new `main`.
+7. A persistent `bevy-main` branch of the repo is maintained and with a dependency on the `main` branch of `bevy` to ensure that we can test out new changes and refine the code before new features and breaking changes go live and reduce release crunch.
+
+### Quick Start examples
+
+### Breakout
+
+**Genres:** arcade
+**Focus:** basic physics, keyboard controls, sprites
+
+Control a paddle with your keyboard to smash bricks and score points in this classic arcade game!
+
+### Falling Sand
+
+**Genres:** simulation, RTS, factory-builder
+**Focus:** advanced ECS patterns, pixel graphics, handling mouse input
+
+Simulate the interactions between different materials as you build your own creation.
+
+### 3D Puzzle Game
+
+**Genres:** puzzle, 3rd person platformer
+**Focus:** camera controls, rule checking, level creation, 3D graphics
+
+Create devious puzzles for your players to solve as they navigate a 3D world.
+
+### Bevy Book chapters
+
+Advanced chapters are skippable, rely on prior knowledge and are not useful for beginners.
+
+1. Getting Started
+   1. Installing Rust and Bevy
+   2. `App`, `AppBuilder` and `World`
+   3. Advanced: Fast compiles
+2. Entities, Components and Systems
+   1. Entities have components
+   2. Systems access data through queries
+   3. Resources are global singletons
+   4. Bundles collect related components
+   5. Spawning, despawning and modifying entities with `Commands`
+   6. Reliable change detection
+   7. Advanced: generic systems
```

Comment:
I'm having second thoughts about explicit "advanced" labels :smile: 

Its a bit arbitrary, and doesn't really solve the "flow" problem when categories have many non-advanced features (or many advanced features). Maybe instead we should just let each consecutive subsection increase in complexity / difficulty naturally (ex: easy / foundational concepts first, fringe / complicated topics last).

Sorting by topic breaks "linear book flow", but I think it gives users more control over how the interact with the book. Users can easily skip topics they don't care about and skip _to_ sections that matter most to them.  A user interested in 2D might skip to that section, see an ECS feature they don't understand, find it in the ECS section and learn it, then skip back to where they were in 2D.

Reply:
> Maybe instead we should just let each consecutive subsection increase in complexity / difficulty naturally (ex: easy / foundational concepts first, fringe / complicated topics last).

I don't mind this idea; we'll just need to be careful about signposting **How to Use This Book** in the introduction so they don't give up after wading through endless minutiae.

I would very much like infoboxes (little square boxes inline or beside the text used for digressions) for advanced asides if we're taking that route. @mockersf, how feasible is that going to be in Zola?

Reply:
Removed for now :) We can come back to this later if needed.

Reply:
> I would very much like infoboxes (little square boxes inline or beside the text used for digressions) for advanced asides if we're taking that route. @mockersf, how feasible is that going to be in Zola?

easy, you'll just have to write the css with flex box to display how you want 😉

Reply:
Eh, I can use footnotes to start and we can migrate to the nicer infoboxes later.